form/src/main/java/org/dhis2/form/ui/provider/inputfield/InputFileProvider.kt

```diff
@@ -23,48 +25,47 @@ internal fun ProvideInputFileResource(
     modifier: Modifier,
     fieldUiModel: FieldUiModel,
     resources: ResourceManager,
+    uiEventHandler: (RecyclerViewUiEvents) -> Unit,
 ) {
-    val uploadState by remember(fieldUiModel) { mutableStateOf(getUploadState(fieldUiModel)) }
+    var uploadState by remember(fieldUiModel) { mutableStateOf(getUploadState(fieldUiModel.displayName, fieldUiModel.isLoadingData)) }
 
-    val fileInputData = fieldUiModel.value?.let {
-        val file = File(it)
-        InputData.FileInputData(
-            fileName = file.name,
-            fileSize = file.length(),
-            filePath = file.path,
-        )
-    }
-
-    val filename by remember(fieldUiModel) {
-        mutableStateOf(fileInputData?.fileName)
-    }
+    val fileInputData =
+        fieldUiModel.displayName?.let {
+            val file = File(it)
+            InputData.FileInputData(
+                fileName = file.path.split("/").last(),
```

Comment:
why it is not enough with `file.name`?

Reply:
With the corrections I was making before it was necessary as the extension wasn't being included, but I have just tested it and it is enough with file.name